File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int cnt=0,curr=-1;
        for(auto& i:nums){
            if(!cnt){
                curr=i;
                cnt=1;
            }
            else if(i==curr){
                cnt++;
            }
            else cnt--;
        }
        return curr;
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        cnt,val = 0,0
        for ele in nums:
            if cnt==0: 
                cnt=1 
                val=ele 
            elif ele==val: 
                cnt+=1 
            else: 
                cnt-=1 
        return val